src/components/MainPage/MainPage.js

```diff
@@ -1,24 +1,158 @@
+import CameraRoll, {
+  type GetPhotosParams,
+  type PhotoIdentifiersPage,
+  type PhotoIdentifier,
+} from '@react-native-community/cameraroll'
 
-import { TouchableButton, PhotoList, } from '../index'
+import { TouchableButton, PhotoList, SelectedPhoto, } from '../index'
+import { MAX_COUNT_LIST_PHOTOS, } from '../../constants'
 import styles from './style'
 
+import { type PhotoDataType, } from '../../types'
+
 type Props = {
   toggleCamera: () => void
 }
 
-const MainPage = ({ toggleCamera, }: Props) => (
-  <View style={styles.container}>
-    <Text style={styles.headline}>Photo List</Text>
-    <PhotoList />
-    <TouchableButton onPress={toggleCamera} style={styles.makePhotoBtn}>
-      <Text style={styles.btnText}>MAKE A PHOTO</Text>
-    </TouchableButton>
-  </View>
-)
+type InitialPhotoState = {
+  photoList: ?Array<PhotoIdentifier>,
+  nextChunksIndicator: ?string
+}
+
+const INITIAL_PHOTO_STATE = {
+  photoList: null,
+  nextChunksIndicator: null,
+}
+
+const MainPage = ({ toggleCamera, }: Props) => {
+  const [{ photoList, nextChunksIndicator, }, setPhotoList] = useState(
+    INITIAL_PHOTO_STATE
+  );
+  ({ photoList, nextChunksIndicator, }: InitialPhotoState)
+
+  const [selectedPhotoData, setSelectedPhoto] = useState(null);
+  (selectedPhotoData: ?PhotoDataType)
+
+  const [chunkNumber, changeChunkNumber] = useState(0);
+  (chunkNumber: number)
+
+  const checkAndroidPermission = async (): Promise<void> => {
+    try {
+      const permission: string =
+        PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE
+      await PermissionsAndroid.request(permission)
+      Promise.resolve()
+    } catch (error) {
+      Promise.reject(error)
+    }
+  }
+
+  const getPhoto = async (): Promise<PhotoIdentifiersPage> => {
+    const startParams: GetPhotosParams = {
+      first: MAX_COUNT_LIST_PHOTOS * 3,
+      assetType: 'Photos',
+    }
+
+    if (nextChunksIndicator) {
+      startParams.after = nextChunksIndicator
+    }
+    const data: PhotoIdentifiersPage = await CameraRoll.getPhotos(startParams)
+
+    return data
+  }
+
+  const setStartData = async () => {
+    checkAndroidPermission()
+    const photoData: PhotoIdentifiersPage = await getPhoto()
+
+    setPhotoList({
+      photoList: photoData.edges,
+      nextChunksIndicator: photoData.page_info.end_cursor || null,
+    })
+  }
 
-export { MainPage, }
+  useEffect(() => {
+    setStartData()
+  }, [])
+
+  const selectPhoto = (data: PhotoDataType): void => setSelectedPhoto(data)
+
+  const closeSelectedPhoto = (): void => setSelectedPhoto(null)
+
+  const getMorePhoto = async (): Promise<void> => {
+    const photoData: PhotoIdentifiersPage = await getPhoto()
+
+    if (photoList) {
+      setPhotoList({
+        photoList: photoList.concat(photoData.edges),
+        nextChunksIndicator: photoData.page_info.end_cursor || null,
+      })
+    }
+  }
+
+  const showPhotoOrLoad = (showeredPhoto) => {
```

Comment:
Maybe `displayedPhoto` instead of `showeredPhoto`?

Reply:
Display may be better then show. I consider that function names should be a verb

Reply:
fixed

Reply:
@Dekim12 I meant an argument name:
const showPhotoOrLoad = **(displayedPhoto)** => {